src/Microsoft.Health.SqlServer/ManagedIdentityAccessTokenHandler.cs

```diff
@@ -1,29 +1,35 @@
-﻿// -------------------------------------------------------------------------------------------------
+// -------------------------------------------------------------------------------------------------
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
 using System.Threading;
 using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
 using EnsureThat;
-using Microsoft.Azure.Services.AppAuthentication;
 
 namespace Microsoft.Health.SqlServer;
 
 public class ManagedIdentityAccessTokenHandler : IAccessTokenHandler
 {
-    private readonly AzureServiceTokenProvider _azureServiceTokenProvider;
+    private readonly string _managedIdentityClientId;
 
-    public ManagedIdentityAccessTokenHandler(AzureServiceTokenProvider azureServiceTokenProvider)
+    public ManagedIdentityAccessTokenHandler(string managedIdentityClientId)
     {
-        EnsureArg.IsNotNull(azureServiceTokenProvider, nameof(azureServiceTokenProvider));
+        EnsureArg.IsNotNull(managedIdentityClientId, nameof(managedIdentityClientId));
 
-        _azureServiceTokenProvider = azureServiceTokenProvider;
+        _managedIdentityClientId = managedIdentityClientId;
     }
 
-    /// <inheritdoc />
-    public Task<string> GetAccessTokenAsync(string resource, CancellationToken cancellationToken)
+    public async Task<string> GetAccessTokenAsync(string resource, CancellationToken cancellationToken)
     {
-        return _azureServiceTokenProvider.GetAccessTokenAsync(resource, cancellationToken: cancellationToken);
+        EnsureArg.IsNotNullOrEmpty(resource, nameof(resource));
+
+        ManagedIdentityCredential credential = new ManagedIdentityCredential(_managedIdentityClientId);
```

Comment:
Should we create this in the `ctor` and save it as a member instead?

Reply:
We can but this also works. Also curious what is the downside of doing it this way?